[docs] [explore] new page for covering indexes (#11764)
yugabyte/yugabyte-db: docs/content/latest/explore/indexes-constraints/covering-index-ysql.md

```sql
EXPLAIN ANALYZE SELECT * FROM demo WHERE upper(username)='NUMBER42';
```

```output
                                                 QUERY PLAN
------------------------------------------------------------------------------------------------------
 Seq Scan on demo  (cost=0.00..105.00 rows=1000 width=40) (actual time=15.694..16.086 rows=1 loops=1)
   Filter: (upper(username) = 'NUMBER42'::text)
   Rows Removed by Filter: 999
 Planning Time: 0.238 ms
 Execution Time: 16.255 ms
(5 rows)
```

- Optimize the SELECT query by creating a functional index as follows:

```sql
CREATE INDEX demo_upper ON demo( (upper(username)) );
```

```sql
EXPLAIN ANALYZE SELECT upper(username) FROM demo WHERE upper(username)='NUMBER42';
```

```output
                                                    QUERY PLAN
-------------------------------------------------------------------------------------------------------------------
 Index Scan using demo_upper on demo  (cost=0.00..5.28 rows=10 width=32) (actual time=2.899..2.903 rows=1 loops=1)
   Index Cond: (upper(username) = 'NUMBER42'::text)
 Planning Time: 13.392 ms
 Execution Time: 3.429 ms
(4 rows)
```

> Using a function based index ([expression index](../../indexes-constraints/expression-index-ysql/)) enables faster access to the rows requested in the query. The problem is that the query planner just takes the expression, sees that there's an index on it, knows that you'll select the `username` column and apply a function on it. Then it thinks that it needs the `username` column without realizing it already has the value with the function applied. In this case, an index only scan covering the column to the index can optimize the query performance.

Reviewer: ```suggestion
Using an expression index enables faster access to the rows requested in the query. The problem is that the query planner just takes the expression, sees that there's an index on it, and knows that you'll select the `username` column and apply a function to it. It then thinks it needs the `username` column without realizing it already has the value with the function applied. In this case, an index only scan covering the column to the index can optimize the query performance.
```

{{< note title="Note" >}}

> For simplicity, the `username` column is used along with the INCLUDE keyword to create the covering index. Generally, a covering index allows a user to perform an index-only scan if the select list in the query matches the columns that are included in the index and additional columns for the index are specified using the INCLUDE keyword.

Reviewer: ```suggestion
For simplicity, the `username` column is used with the INCLUDE keyword to create the covering index. Generally, a covering index allows you to perform an index-only scan if the query select list matches the columns that are included in the index and the additional columns added using the INCLUDE keyword.
```

> Ideally, use columns that are updated frequently in the INCLUDE clause. But for other cases, it would probably be faster to just have an index on the key columns.

Reviewer: ```suggestion

Ideally, specify columns that are updated frequently in the INCLUDE clause. For other cases, it is probably faster to index all the key columns.
```

{{< /note >}}

> - Create a covering index by including the username column along with the INCLUDE keyword.

Reviewer: ```suggestion
- Create a covering index by specifying the username column in the INCLUDE clause.
```
